src/main/markdown/doc/latest/DevGuideCodingBasicsJsinterop.md

```diff
@@ -0,0 +1,228 @@
+
+```
+@JsType(isNative=true, namespace=GLOBAL)
+public class JSON {
+  public native static String stringify(Object obj);
+  
+  public native static Object parse(String obj);
+}
+```
+
+additionally the name property can be used to export or import an object with a different name than the Java variable name. `@JsType(name="newName")`
+
+annotating a class @JsType is equivalent to annotating all its methods with @JsMethod and all its properties with @JsProperty, so no need to add them explicity.  
+
+<ul>Important notes: 
+    
+  <li> constructors of native imported objects should be empty and cannot contain any statement (checked at compile time)</li>
+  <li> native objects cannot have non native methods, unless annotated with @JsOverlay (checked at compile time)</li>
+  <li> native objects cannot have assigned properties, (int x = 10 should be int x) </li>
+  <li> @JsType is not transitive. Child objects need to be annotated as well. </li>
+</ul>
+
+
+## @JsProperty:
+
+@JsProperty is used to annotate a static class property or a property whose class has a @JsConstructor annotation, to be exported/import to/from javascript, without having to expose all the object fields and methods (without @JsType)
+
+
+```
+package com.gwt.example;
+
+  
+public class myClass {
+
+@Jsproperty
+public static String name = "foo";
+
+public myClass(String name){
+this.name = name;
+}
+
+}
+```
+
+
+```
+//the package name serves as JS namespace
+com.gwt.example.myClasse.name;//foo
+```
+
+## @JsMethod:
+
+@JsMethod is used to annotate static methods that need to exposed to js without having to expose the object itself
+(without @JsType). 
+
+```
+package com.gwt.example;
+
+public class Record {
+  
+  
+  public Record() {
+    
+  }
+  
+  @JsMethod(namespace="hello.world")
+  public static int test(int i) {
+    
+    return i + 10;
+  }
+  
+}
+```
+```
+hello.world.test(20); //Results in 30
+
+var record = new com.gwt.example.Record(); 
+
+//throws error at run time
+Uncaught ReferenceError: com is not defined
+    at <anonymous>:1:18
+```
+
+If the method is not static then @JsMethod can work if the object can be instantiated from javascript. This can be achieved using @JsConstructor. 
+
+## @JsConstructor:
+
+`@JsConstructor` can be used to expose an object constructor. It can be used along with `@JsProperty` and `@JsMethod` to expose only certain fields or methods. For example: 
+
+```
+package com.gwt.example;
+
+
+public class myClass {
+
+@JsProperty
+public String name;
+
+
+public String id = bar;
+
+@JsConstructor
+public myClass(String name){
+this.name = name;
+}
+
+@JsMethod
+public void sayHello(){
+return 'Hello' + this.name;
+}
+
+
+public void sayGoodbye(){
+return 'Bye ' + this.name;
+}
+}
+```
+
+This results in the method `sayHello()` and the property `name` to be exposed. The method `sayGoodbye()` and the property `id` will not be accessible from a javascript script. 
```

Comment:
Also, the type can be constructed from JS using `new com.gwt.example.myClass(name)`: it is the `@JsConstructor` that exposes the JS constructor, not the `@JsType` (AFAICT). This is why, except for its `isNative`, `@JsType` is only a shortcut for applying `@JsConstructor`, `@JsMethod` and `@JsProperty` on the type members.